### ISSUE TITLE
Torch: gradient_clip wrong when grad_scaler is used

### ISSUE THREAD

**Issue by michelwi** [rwth-i6/returnn#1590]
With torch AMP by default we get a gradient scaler to prevent underflow of (b)float16 gradients. This interacts with gradient clipping.

In [this example](https://pytorch.org/docs/stable/notes/amp_examples.html#working-with-unscaled-gradients) they use unscaled gradients for clipping, whereas [in returnn](https://github.com/rwth-i6/returnn/blob/cf394a24cc3d9491cf9259f342d04f6a29dda2ec/returnn/torch/updater.py#L180) the scaled gradient seem to be used.

So in principle the gradient_clip_global_norm would already remove the constant scaling factor (assuming that `scale * norm(gradient) >> gradient_clip_global_norm`) and the `grad_scaler.step(self.optimizer)` would again unscale the gradients and thereby further reduce the magnitude.
Or inversely, the effective value gradient_clip_global_norm with AMP is reduced by a factor of 65k..
